src/components/HideCheckbox.tsx

```diff
@@ -1,47 +1,42 @@
-import React, { useContext, useMemo } from 'react';
+import React, { useContext } from 'react';
 import { Checkbox, Tooltip } from 'antd';
 
-import { CheckboxState } from '../constants';
+import { CheckboxProps } from '../types';
 import { VisibilityContext } from '../AgentVisibilityContext';
-import { UIDisplayEntry } from '@aics/simularium-viewer/type-declarations/simularium/SelectionInterface';
+import { CheckboxState, tooltipMap } from '../constants';
+import { getChildren } from '../utils';
 
-interface HideCheckboxProps {
-  agent: UIDisplayEntry;
-}
-
-const HideCheckbox: React.FunctionComponent<HideCheckboxProps> = (
-  props: HideCheckboxProps
+const HideCheckbox: React.FunctionComponent<CheckboxProps> = (
+  props: CheckboxProps
 ): JSX.Element => {
   const { agent } = props;
-
-  const { handleVisibilityCheckboxChange, hiddenAgents } =
+  const { handleHideCheckboxChange, hiddenAgents } =
     useContext(VisibilityContext);
 
-  const getHideCheckboxStatus = (): CheckboxState => {
-    if (hiddenAgents[agent.name]?.length === 0) {
+  const selections = hiddenAgents[agent.name];
```

Comment:
it feels to me like selections should be a prop of all these components. You could also make the `handleHideCheckboxChange(agent.name, children` at the higher level and then this component wouldn't have to access the context at all. but it's just a preference, not a blocking issue. My thinking is that you access the context once in `AgentRow` and then send all the necessary information into each of the subcomponents since they're already receiving some props. but if you think it's clearer this way it's fine